components/nimbus/src/evaluator.rs

```diff
@@ -58,10 +66,20 @@ pub fn evaluate_enrollment(
         });
     }
 
+    let is_already_enrolled = if let Some(status) = previous_enrollment {
+        status.is_enrolled()
+    } else {
+        false
+    };
+
     // Get targeting out of the way - "if let chains" are experimental,
     // otherwise we could improve this.
     if let Some(expr) = &exp.targeting {
-        if let Some(status) = targeting(expr, app_context) {
+        let targeting_attributes = TargetingAttributes {
+            app_context: app_context.clone(),
```

Comment:
Hmmm I would have to have the TargetingAttributes keep a reference to the `AppContext` which should work since the lifetime of the TargetingAttributes is pretty small compared to the `AppContext`... 

I'll keep it like this for now, since I mutate the inner `AppContext` in the tests but it could be a good clean-up for the future.. I'll file a small backlog ticket